app/Models/Relations/EntityRelations.php

```diff
@@ -238,7 +238,7 @@ public function tags()
     public function tagsWithEntity()
     {
         if ($this->tagsWithEntity === false) {
-            $this->tagsWithEntity = $this->tags()->with('entity')->get();
+            $this->tagsWithEntity = $this->tags()->with('entity')->where('is_hidden', '=', '0')->get();
```

Comment:
Since we're doing a `where()` on the db, we need to index the field to avoid bad performances.